packages/contracts/contracts/CollateralManager.sol

```diff
@@ -0,0 +1,372 @@
+// SPDX-Licence-Identifier: MIT
+pragma solidity >=0.8.0 <0.9.0;
+
+// Contracts
+import "@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+// Libraries
+import "@openzeppelin/contracts-upgradeable/utils/structs/EnumerableSetUpgradeable.sol";
+
+// Interfaces
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/IERC721Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC1155/IERC1155Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/IERC721ReceiverUpgradeable.sol";
+import "./TellerV2.sol";
+
+contract CollateralManager is OwnableUpgradeable, ICollateralManager {
+    using EnumerableSetUpgradeable for EnumerableSetUpgradeable.AddressSet;
+    TellerV2 public tellerV2;
```

Comment:
can this be an interface ?  I think it would significantly reduce the amount of bytecode associated with this contract file if so.